### DISCUSSION TITLE
Is it possible to add cluster host as node?

### DISCUSSION THREAD

**Discussion by tukusejssirs** [gravitational/teleport#22269] [Q&A] [answered]
I’d like to add cluster host as node, so that I could `tsh ssh` to it. Is it possible?

**Comment by webvictim** [comment, marked as answer]
Yes. Set this in your `/etc/teleport.yaml` config and restart Teleport:

```
ssh_service:
  enabled: true
```

**Reply by tukusejssirs** [reply]
Well, I have `ssh_service` enabled already, but the cluster `nodename` is not listed in the servers/nodes list. And when I try to connect to it using `tsh ssh $user@$nodename`, it fails with the following error:

```bash
ERROR: failed connecting to node $nodename. Teleport proxy failed to connect to "node" agent "$nodename:3022" over direct dial:

  dial tcp: lookup $nodename: Temporary failure in name resolution

This usually means that the agent is offline or has disconnected. Check the
agent logs and, if the issue persists, try restarting it or re-registering it
with the cluster.
```

I can `telnet` on `3022` of the domain where the cluster is running.

---

Should I add another service running `teleport` as a node with separate config from the cluster config?

**Reply by tukusejssirs** [reply]
Okay, the issue is that I ran Teleport cluster in a Docker container.

Two possible solutions:
- run Teleport cluster directly on the host;
- keep Teleport cluster running in a Docker container and add a Teleport node service running directly on the host.
